pydantic/types.py

```diff
@@ -838,3 +842,97 @@ def to(self, unit: str) -> float:
             raise errors.InvalidByteSizeUnit(unit=unit)
 
         return self / unit_div
+
+
+if not TYPE_CHECKING:  # noqa: C901
+    # Actual implementation, Cythonable
+    class SplitStr(list):
+        # Needed for pydantic to detect that this is a list
+        __origin__ = list
+        __args__: List[Type[T]]
+        item_type: Type[T]
+
+        # classmethod not required, but make Cython happy until 0.30.0 is available
+        # ref: https://github.com/cython/cython/issues/3764
+        @classmethod
+        def __class_getitem__(cls, item_type: Type[T]) -> Type[List[T]]:
+            # __args__ is needed to conform to typing generics api
+            namespace = {'item_type': item_type, '__args__': [item_type]}
+            # We use new_class to be able to deal with Generic types
+            return new_class(f'{cls.__name__}Type', (cls,), {}, lambda ns: ns.update(namespace))
+
+        @classmethod
+        def __get_validators__(cls) -> 'CallableGenerator':
+            yield cls.split_str_validator
+
+        @classmethod
+        def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
+            str_schema = {**field_schema, 'type': 'string'}
+            if 'items' in str_schema:
+                del str_schema['items']
+            any_of = [field_schema.copy(), str_schema]
+            field_schema.clear()
+            field_schema['anyOf'] = any_of
+
+        @classmethod
+        def split_str_validator(cls, v: 'Optional[Union[List[T], str]]', field: 'ModelField') -> 'Optional[List[T]]':
+            if v is None and not field.required:
+                return None
+            if isinstance(v, str):
+                use_v: List[Any] = v.split()
+            else:
+                use_v = list_validator(v)
+            return use_v
+
+    class SpaceSeparated(SplitStr):
+        pass
+
+    class CommaSeparated(SplitStr):
+        @classmethod
+        def split_str_validator(cls, v: 'Optional[Union[List[T], str]]', field: 'ModelField') -> 'Optional[List[T]]':
+            if v is None and not field.required:
+                return None
+            if isinstance(v, str):
+                use_v: List[Any] = v.split(',')
```

Comment:
I think we need more advanced logic here (and in the case of splitting on spaces) (or at least a prominent warning).

What happens if I enter `apples, apples and pears, "apples, pears and bananas"`? Should we get 
    ['apples', 'apples and pears', 'apples, pears and bananas']
or
    ['apples', ' apples and pears', ' "apples', ' pears and bananas"']

as this logic would produce? I would imagine many people would expect the first.

We could easily escape commas with quotes, but then we need to be able to escape quotes, and we've basically got to csv parsing.

I'm not sure what to do here. Can we easily use the `csv` module to do the heavy lifting here?

Reply:
It looks [like](https://github.com/python/cpython/blob/5c0eed7375fdd791cc5e19ceabfab4170ad44062/Modules/_csv.c) `csv` is actually implemented in c so should be pretty fast.